appinventor/blocklyeditor/src/blocks/text.js

```diff
@@ -32,14 +30,19 @@ Blockly.Blocks['text'] = {
     this.setOutput(true, AI.BlockUtils.YailTypeToBlocklyType("text", AI.BlockUtils.OUTPUT));
     this.setTooltip(Blockly.Msg.LANG_TEXT_TEXT_TOOLTIP);
   },
-  mutationToDom: function () {
-    const container = document.createElement('mutation');
-    container.setAttribute('output', JSON.stringify(this.outputConnection.getCheck()));
-    return container;
+  inputValidator: function (newVal) {
+    var block = this.getSourceBlock();
+    if (block.workspace.isLoading) {
+      Blockly.Blocks.text.setOutputOnFinishEdit.bind(block)(newVal);
```

Comment:
I wonder if it is more efficient to call `.call(block, newVal)` rather than creating a new bound function for every text. Also, since the functions are copied to the block object couldn't we just do `this.setOutputOnFinishEdit(...)`?